spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -290,9 +290,22 @@ def to_s
   end
 end
 
-RSpec.describe "expect { ... }.not_to raise_error(message)" do
+RSpec.describe "expect { ... }.not_to raise_error('message')" do
   it "issues a warning" do
-    expect_warning_with_call_site __FILE__, __LINE__+1, /risks false positives/
+    expect_warning_with_call_site __FILE__, __LINE__+1, /not_to raise_error\(message\)` risks false positives/
+    expect { raise 'blarg' }.not_to raise_error("blah")
+  end
+
+  it "can supresses the warning when configured to do so", :warn_about_potential_false_positives do
```

Comment:
Wondering why ‘warn_about_potential_false_positives’ metadata is needed here to warn, but it warns anyway in other examples? Is it to restore the setting to the default? It just reads confusingly when used this way.

Reply:
I see that we do the same in other examples in this file.

Reply:
I didn't know about this until you asked (and I wasn't sure until I tested it out), but adding a shared context like [this one](https://github.com/rspec/rspec-expectations/blob/main/spec/spec_helper.rb#L120-L124) (with a tag specified) causes that context to be automatically included in any context that supplies that same tag. So including this metadata causes an automatic cleanup to occur. I, of course, had just copied it from other tests without thinking :-)